x-pack/plugins/ml/server/shared_services/shared_services.ts

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { MlServerLicense } from '../lib/license';
+
+import { SpacesPluginSetup } from '../../../spaces/server';
+import { CloudSetup } from '../../../cloud/server';
+import { licenseChecks } from './license_checks';
+import { MlSystemProvider, getMlSystemProvider } from './providers/system';
+import { JobServiceProvider, getJobServiceProvider } from './providers/job_service';
+import { ModulesProvider, getModulesProvider } from './providers/modules';
+import { ResultsServiceProvider, getResultsServiceProvider } from './providers/results_service';
+import {
+  AnomalyDetectorsProvider,
+  getAnomalyDetectorsProvider,
+} from './providers/anomaly_detectors';
+
+export type SharedServices = JobServiceProvider &
+  AnomalyDetectorsProvider &
+  MlSystemProvider &
+  ModulesProvider &
+  ResultsServiceProvider;
+
+export function createSharedServices(
+  mlLicense: MlServerLicense,
+  spaces: SpacesPluginSetup | undefined,
+  cloud: CloudSetup
+): SharedServices {
+  const { isFullLicense, isMinimumLicense } = licenseChecks(mlLicense);
+
+  return {
```

Comment:
Would it make sense to give these a consistent naming style, so for example `getJobProvider` and `getResultsProvider`, rather than having `Service` in some of them and not in others?

Reply:
i've kept the names the same as their related routes:
```
job_service.ts
anomaly_detectors.ts
system.ts
modules.ts
results_service.ts
```